### DISCUSSION TITLE
github pr discussion

### DISCUSSION THREAD

**Discussion by amitkumar1834** [community/community#55761] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

while raising a pr from dev to qa i want my changes to reflect in the pr not any other changes from other team members how can i do that?

**Comment by airtower-luna** [comment]
If you want to merge your changes without your colleagues', don't develop on the same branch.

If you need to move a few commits over [cherry-picking](https://git-scm.com/docs/git-cherry-pick) them might be an option, but that's a very different thing from a PR.

**Reply by amitkumar1834** [reply]
i am developing on diff branch but while raising a PR from DEV to QA i want only changes to go just wanted to know if its possible in anyway?

**Reply by airtower-luna** [reply]
When you merge two branches, you cannot merge only some of the commits. If you want to merge only certain commits into the target branch, you need to merge a branch that has *only those commits*.

**Reply by amitkumar1834** [reply]
Suppose i am changing a line number 33 and my team member changed line 23 and he pushed the code to dev branch from his branch....
and i am pushing my changes to dev branch after him.....so when a pr is raised from dev to qa only my changes in line 33 is show in the pr is it possible?

**Reply by airtower-luna** [reply]
> so when a pr is raised from dev to qa only my changes in line 33 is show in the pr is it possible?

No, as I already wrote. If you merge `dev`, you'll get *all* commits that are on that branch, yours *and* your colleagues'.

If you want to merge separately, you have to do it from a branch that contains *only* your changes (merge your own branch directly to `qa`, *not* via `dev`). That's why it's usually good to work with feature branches, and not a shared `dev` branch.

If you need to carry over a fix and don't mind creating duplicate commit, look at `git cherry-pick` (as I mentioned above).

**Reply by amitkumar1834** [reply]
okay i have another scenario i dont want some of files to go to qa which are not necessary those changes are done by my other team member how can i do that?
